### DISCUSSION TITLE
needing help on authenticating when git cloning a respository?

### DISCUSSION THREAD

**Discussion by drfreckleshitz** [community/community#84762] [Code Security]
### Select Topic Area

General

### Body

I was learning how to git clone a repository on my mac terminal  and was asking me for my git hub username,but the problem comes when putting in the password and wont let me type it but instead shows me this,what should i do?
<img width="744" alt="Screenshot 2024-01-06 at 4 06 01 PM" src="https://github.com/community/community/assets/103446957/bb91acbb-91e2-42f9-a04c-83a7085fa2c3">

**Comment by bonizario** [comment]
Your password may be typed without visual feedback like "*" symbols. Hit enter and it should be enough to log in.

You can avoid being prompted for your password by configuring Git to cache your credentials.

But what I would recommend is generating a new SSH key and adding it to the `ssh-agent` instead.

See: https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent
